packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsBlocklistSection.tsx

```diff
@@ -26,31 +32,51 @@ export const SettingsAccountsBlocklistSection = () => {
     objectNameSingular: CoreObjectNameSingular.Blocklist,
   });
 
+  const { updateOneRecord: updateBlocklistEmail } =
+    useUpdateOneRecord<BlocklistItem>({
+      objectNameSingular: CoreObjectNameSingular.Blocklist,
+    });
+
   const handleBlockedEmailRemove = (id: string) => {
     deleteBlocklistItem(id);
   };
 
-  const updateBlockedEmailList = (handle: string) => {
+  const addNewBlockedEmail = (contact: BlocklistItem) => {
     createBlocklistItem({
-      handle,
+      levels: contact.levels,
+      handle: contact.handle,
       workspaceMemberId: currentWorkspaceMember?.id,
     });
   };
 
+  const updateBlockedEmail = (contact: BlocklistItem) => {
+    updateBlocklistEmail({
+      idToUpdate: contact.id,
+      updateOneRecordInput: {
+        levels: contact.levels,
+        handle: contact.handle,
+      },
+    });
+  };
+
   return (
     <Section>
       <H2Title
         title="Blocklist"
         description="Exclude the following people and domains from my email sync"
       />
-      <SettingsAccountsBlocklistInput
-        blockedEmailOrDomainList={blocklist.map((item) => item.handle)}
-        updateBlockedEmailList={updateBlockedEmailList}
-      />
-      <SettingsAccountsBlocklistTable
-        blocklist={blocklist}
-        handleBlockedEmailRemove={handleBlockedEmailRemove}
-      />
+      <BlocklistContext.Provider
```

Comment:
I'm not sure we need a Context + handle the updateOneRecord in the parent component. Why not dealing with it directly in the row? This would avoid some complexiity

Reply:
100% agree, I made it jic we extend it in the future